GitHub: Improve Dependabot.

Gacko opened the pull request:
<!--- Provide a general summary of your changes in the Title above --->
<!--- Please don't @-mention people in PR or commit messages (do so in an additional comment). --->
<!--- Please make sure you title is descriptive, it is used in the Release notes to let others know what it does ---> 

## What this PR does / why we need it:
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

This morning I found out that Dependabot fails to update our Dockerfiles:

https://github.com/kubernetes/ingress-nginx/network/updates/892831825

It's looking for a Dockerfile in the `images` directory. Apparently the current config is not working recursively. After a short research I found that the `directories` (not `directory`) config key supports wildcards and nested locations:

https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#directories

At first I thought using just `*` would be enough, but reading the docs a second time, I came to the conclusion that this would only cover the root directory and directories in side the root directory, but not nested directories. So I went for `/` and `**/*` to cover both the root directory and nested directories.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] CVE Report (Scanner found CVE and adding report)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation only

## Which issue/s this PR fixes
<!--
(optional, in `fixes #<issue number>` format, will close that issue when PR gets merged):

fixes #
-->

https://github.com/kubernetes/ingress-nginx/network/updates/892831825

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

Cannot be tested before merging.

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/kubernetes/ingress-nginx/blob/main/CONTRIBUTING.md) guide
- [ ] I have added unit and/or e2e tests to cover my changes.
- [ ] All new and existing tests passed.

Gacko commented:
/triage accepted
/kind bug
/priority important-soon
/cc @strongjz @rikatz @cpanato @tao12345666333

Gacko commented on .github/dependabot.yml:
```diff
@@ -12,9 +13,12 @@ updates:
     groups:
       all:
         update-types:
+          - "minor"
```
Ah, ok. I was not sure why we did not add `minor` there, also because it only creates a PR we still can reject, right?

cpanato replied:
yes but then it will not open new one, only if a new version comes up, and then you will need to open manually which will be work that we try to eliminate using dependabot

Gacko replied:
Ah, sure, sure, makes sense!

Gacko commented:
Also @cpanato, as you seem to be around: What's your understanding of the docs? As stated above I'm not sure if just `*` would be enough as, according to the docs and my understanding, it only covers the root directory and 1st level directories, but not nested ones. WDYT?

cpanato commented:
> Also @cpanato, as you seem to be around: What's your understanding of the docs? As stated above I'm not sure if just `*` would be enough as, according to the docs and my understanding, it only covers the root directory and 1st level directories, but not nested ones. WDYT?

yes, that is my understanding as well but i would keep as you did

cpanato commented on .github/dependabot.yml:
```diff
@@ -1,8 +1,9 @@
+  - package-ecosystem: "github-actions"
+    directories:
+      - "/"
+      - "**/*"
```
Do we have other Github actions besides the .github directory?

Gacko replied:
Ah, damn, my fault! For GitHub Actions `/` is fine, of course. Fixing it!

Gacko commented:
Ok, nice!

Sorry for the confusion. I re-ordered the entries a bit. You might want to have a look at the current state instead of the change only: https://github.com/kubernetes/ingress-nginx/blob/2480386ef37576992f5470905797fc3c1d45e540/.github/dependabot.yml

Gacko commented:
I amended the PR to not flood us with updates on merge. Also I split my changes in atomic commits.

Right now I only fixed the directories for the Docker dependencies and added most occurrences of `go.mod` files to the Go section.

Before Dependabot failed to update Dockerfiles because they were not lying in `/images` but in `/rootfs` & `/images/*/rootfs`. This is why I just went for `**/rootfs` as otherwise we would need separate entries for `/rootfs` and `/images/*/rootfs`.

The same basically applies for Go modules: We have `go.mod` files in `/` & `/images/*/rootfs`.

We can still broaden the scope later, but for now I would like to do just a small first step to not get flooded by dependency bump PRs and possible breaking changes.

cpanato commented on .github/dependabot.yml:
```diff
@@ -26,16 +27,17 @@ updates:
       - "ok-to-test"
     groups:
-      actions:
+      all:
```
This is the name that will be included in the PR, so we can call this `gomod` and update the others as well to use meaningful names

Gacko replied:
Yeah, sure, sounds good!

Gacko commented:
/cherry-pick release-1.10

Gacko commented:
/cherry-pick release-1.11

cpanato reviewed (approved):
thanks

/lgtm
